### PR TITLE
Master job and bigsky

### PR DESCRIPTION
**Pull request by rroutsong**
Major change from pip-centric packaging to git submodule packaging. Will need to update submodule once that PR is merged.

Address https://github.com/OpenOmics/Dmux/issues/9 https://github.com/OpenOmics/Dmux/issues/8 https://github.com/OpenOmics/Dmux/issues/7 https://github.com/OpenOmics/Dmux/issues/13 and https://github.com/OpenOmics/Dmux/issues/16

Given --dry_run or --local, jobs are not submitted to slurm. 

Attached is example dry run output.

Example output on biowulf: `/data/RTB_GRS/dev/Dmux/test_ngsqc2`
Example output on bigsky: `/gs1/Scratch/routsongrm/dev/Dmux/test_dmux`

Dry run example output is attached [here](https://github.com/OpenOmics/Dmux/files/13368111/test_dry_run.txt).

### CODE REVIEW

File: .gitmodules
Comment (skchronicles) [inline .gitmodules:1]: Thank you for adding this!

File: .tests/ngs_qa_qc/WT_S1.R1.fastq.gz
Comment (skchronicles) [inline .tests/ngs_qa_qc/WT_S1.R1.fastq.gz:1]: I usually keep these for dry-running the pipeline but if we plan to support mutually exclusive mixed inputs (BCL vs. FastQ file) we may want to add these back.

File: INSTALL.md
Comment (skchronicles) [inline INSTALL.md:1]: We may want to add more detailed instructions for users setting up or installing the pipeline.

File: docker/NGS_QC/fastq_screen.conf
Comment (skchronicles) [inline docker/NGS_QC/fastq_screen.conf:1]: This file should ideally be created on the fly, or we need to provide instructions for setting up the pipeline on other clusters.

File: docker/demux/bcl-convert-3.6.3-1.el7.x86_64.rpm
Comment (skchronicles) [inline docker/demux/bcl-convert-3.6.3-1.el7.x86_64.rpm:1]: Did you mean to commit this file? It's pretty large (>1MB).

**Review by skchronicles — CHANGES_REQUESTED** [review]
Hey @rroutsong,

Everything looks great! I just have a few minor comments. The only thing I think we may need to change or think about before merging everything is the commit of the 5MB RPM package for BCL convert. We may need to permanently remove this from the git commit history as it will dramatically increase the size of the project's commit history.

File: INSTALL.md
Comment (rroutsong) [reply INSTALL.md:1]: Yeah there is a good amount of documentation I need to update. I can work that in to a single PR next.

File: docker/NGS_QC/fastq_screen.conf
Comment (skchronicles) [reply docker/NGS_QC/fastq_screen.conf:1]: Okay, perfect, thank you for the clarification. That sounds good!

File: .tests/ngs_qa_qc/WT_S1.R1.fastq.gz
Comment (skchronicles) [reply .tests/ngs_qa_qc/WT_S1.R1.fastq.gz:1]: No worries, we can always touch those files in the ci prior to the dry-run later. Let's keep it like this for now.

**Review by skchronicles — APPROVED** [review]
Thank you for addressing my comments and questions! Everything looks great. Thank you for the updates!

**Pull request merged**